src/components/vendors/VendorSelector.tsx

```diff
@@ -63,7 +63,7 @@ export function VendorSelector(props: VendorSelectorProps) {
           onDismiss={props.onClearButtonClick}
           initiallyVisible={props.initiallyVisible}
           sortBy="name|asc"
-          staleTime={props.staleTime || 500}
+          staleTime={props.staleTime || Infinity}
```

Comment:
@beganovich @turbo124 The same reason, which is commented on in the Client Selector, is also the reason for this change.